pkg/operator/controller/ingress/controller.go

```diff
@@ -473,17 +473,10 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 			}
 			switch lbType {
 			case operatorv1.AWSLoadBalancerProvider:
-				// The only provider parameter that is supported
-				// for AWS is the connection idle timeout for
-				// classic ELBs.
-				var specIdleTimeout, statusIdleTimeout metav1.Duration
-				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
-					specIdleTimeout = specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
-				}
-				if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
-					statusIdleTimeout = statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
-				}
-				if specIdleTimeout != statusIdleTimeout {
+				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
+					statusLB.ProviderParameters = specLB.ProviderParameters
```

Comment:
Interesting, this looks a lot cleaner. So we didn't need to check if the lbStatus's ProviderParameters to see if it was empty or a `AWSClassicLoadBalancer`?

Reply:
The else part will take of this.